packages/client/hmi-client/src/components/model/petrinet/tera-initial-entry.vue

```diff
@@ -30,28 +30,28 @@
 </template>
 
 <script setup lang="ts">
-import { ModelConfiguration } from '@/types/Types';
-import {
-	getInitialName,
-	getInitialDescription,
-	getInitialUnit,
-	getInitialExpression,
-	getInitialSource
-} from '@/services/model-configurations';
+import { Model, ModelConfiguration } from '@/types/Types';
+import { getInitialExpression, getInitialSource } from '@/services/model-configurations';
 import TeraInput from '@/components/widgets/tera-input.vue';
 import { ref } from 'vue';
 import Button from 'primevue/button';
+import {
+	getInitialDescription,
+	getInitialName,
+	getInitialUnits
+} from '@/model-representation/service';
 
 const props = defineProps<{
+	model: Model;
```

Comment:
If given the model configuration already, why do you also need to be given the model?
Im not sure if i am misunderstanding the new structure

Reply:
The exercise to revamp ModelConfiguration is to make a structured way to store any related metadata to the ModelConfiguration.  Its a better solution than shoving everything into the `metadata` key and overall to make it easier for us to work with a ModelConfiguration without explicitly sticking to the AMR data structure.

The use case here for needing Model values is for the fields (name, description, unit), is cause they are not stored on the ModelConfiguration, they are taken from the Model.  These fields are unique to the Model not ModelConfiguration.

Reply:
after quick chat:
model config has a model ID not a model.
We do not want to make GET calls